options/configurable.cc

```diff
@@ -0,0 +1,971 @@
+  const std::string& option_value =
+      input_strings_escaped ? UnescapeOptionString(value) : value;
+  bool found_it = false;
+  Status status;
+#ifndef ROCKSDB_LITE
+  status = SetOption(db_opts, option_name, option_value, input_strings_escaped,
+                     &found_it);
+#endif
+  if (!found_it) {
+    status = SetUnknown(db_opts, option_name, option_value);
+  }
+  return status;
+}
+
+/**
+ * Looks for the named option amongst the options for this type and sets
+ * the value for it to be the input value.
+ * If the name was found, found_option will be set to true and the resulting
+ * status should be returned.
+ */
+#ifndef ROCKSDB_LITE
+Status Configurable::SetOption(const DBOptions& db_opts,
+                               const std::string& name,
+                               const std::string& value,
+                               bool input_strings_escaped, bool* found_option) {
+  // By the time this method is called, the name is the short name (no prefix)
+  // and the value is unescaped.
+  Status status;
+  *found_option = false;
+  // Look for the name in all of the registered option maps until it is found
+  for (auto iter : options_) {
+    const auto& opt_map = iter.second.second;
+    void* opt_ptr = iter.second.first;
+    // Look up the value in the map
+    auto opt_iter = FindOption(name, opt_map);
+    if (opt_iter != opt_map.end()) {
+      const auto& opt_info = opt_iter->second;
+      *found_option = true;
+      if (name == opt_iter->first) {
+        status = ParseOption(opt_info, db_opts, opt_ptr, name, value,
+                             input_strings_escaped);
+      } else {
+        status = ParseOption(opt_info, db_opts, opt_ptr,
+                             name.substr(opt_iter->first.size() + 1), value,
+                             input_strings_escaped);
+      }
+      if (status.IsNotFound()) {
+        *found_option = false;
+      }
+      return status;
+    } else {
+      for (auto map_iter : opt_map) {
+        if (map_iter.second.IsConfigurable()) {
+          char* opt_addr = GetOptAddress(map_iter.second, opt_ptr);
+          Configurable* config =
+              map_iter.second.GetPointer<Configurable>(opt_addr);
+          if (config != nullptr) {
+            status = config->ConfigureOption(db_opts, name, value);
+            if (!status.IsNotFound()) {
+              *found_option = true;
+              return status;
+            }
+          }
+        }  // End if config option
+      }
+    }
+  }
+  return status;
+}
+
+Status Configurable::ParseOption(const OptionTypeInfo& opt_info,
+                                 const DBOptions& db_opts, void* opt_ptr,
+                                 const std::string& opt_name,
+                                 const std::string& opt_value,
+                                 bool input_strings_escaped) {
+  Status status;
+  if (opt_info.verification != OptionVerificationType::kDeprecated) {
```

Comment:
Looks like we can do `if (opt_info.verification == OptionVerificationType::kDeprecated)` and return early so as to save some indentation in the remaining part of this function.